src/main/java/com/gmail/llmdlio/townyflight/listeners/EnemyEnterTownListener.java

```diff
@@ -0,0 +1,177 @@
+		if (resident == null)
+			return;
+
+		// Get the town
+		final Town town = TownyAPI.getInstance().getTown(event.getPlayer().getLocation());
+		if (town == null)
+			return;
+
+		if(playerDisablesFlight(town, resident)){
+			api.incrementEnemiesInTown(town);
+		}
+	}
+
+	/*
+	 * Listener for when enemies log out inside a town.
+	 * It will decrement the enemiesInTown counter for the town.
+	 */
+	@EventHandler
+	public void enemyLogOutListener(PlayerQuitEvent event) {
+
+		// Get the resident
+		final Resident resident = TownyAPI.getInstance().getResident(event.getPlayer().getUniqueId());
+		if (resident == null)
+			return;
+
+		// Get the town
+		final Town town = TownyAPI.getInstance().getTown(event.getPlayer().getLocation());
+		if (town == null)
+			return;
+
+		if(playerDisablesFlight(town, resident)){
+			api.decrementEnemiesInTown(town);
+		}
+	}
+
+	/*
+	 * Listener for when enemies teleport from a town.
+	 * It will decrement the enemiesInTown counter for the town.
+	 */
+	@EventHandler(priority = EventPriority.MONITOR)
+	private void enemyTeleports(PlayerTeleportEvent event) {
+		if (event.getCause() != PlayerTeleportEvent.TeleportCause.PLUGIN || event.getCause() != PlayerTeleportEvent.TeleportCause.COMMAND)
+			return;
+
+		Town town = TownyAPI.getInstance().getTown(event.getFrom());
+		if (town == null)
+			return;
+
+		final Resident resident = TownyAPI.getInstance().getResident(event.getPlayer().getUniqueId());
+		if (resident == null)
+			return;
+
+		if(playerDisablesFlight(town, resident)){
+			api.decrementEnemiesInTown(town);
+		}
+	}
+
+	/*
+	 * A helper method which returns true if a player should disable flight depending on config setting.
+	 */
+	public boolean playerDisablesFlight(Town town, Resident resident){
+		if (Settings.flightDisableBy.equals("ALLY")){
+
+			// If they don't belong to a town, treated as a NEUTRAL player
+			if(TownyAPI.getInstance().getResidentTownOrNull(resident) == null){
+				return true;
+			}
+
+			if (!CombatUtil.isSameTown(town, TownyAPI.getInstance().getResidentTownOrNull(resident))) {
+				return true;
+			}
+
+		}
+		if(Settings.flightDisableBy.equals("NEUTRAL")){
+
+			// If they don't belong to a town, treated as a NEUTRAL player
+			if(TownyAPI.getInstance().getResidentTownOrNull(resident) == null){
+				return true;
+			}
+
+			if (!CombatUtil.isSameTown(town, TownyAPI.getInstance().getResidentTownOrNull(resident)) && !CombatUtil.isAlly(town, TownyAPI.getInstance().getResidentTownOrNull(resident))) {
+				return true;
+			}
+		}
+
+		if(Settings.flightDisableBy.equals("ENEMY")) {
+
+			// If they don't belong to a town, treated as a NEUTRAL player
+			if(TownyAPI.getInstance().getResidentTownOrNull(resident) == null){
+				return false;
+			}
+			if (CombatUtil.isEnemy(town, TownyAPI.getInstance().getResidentTownOrNull(resident)) || CombatUtil.isEnemy(TownyAPI.getInstance().getResidentTownOrNull(resident), town)) {
+				return true;
+			}
+		}
+
+		return false;
+	}
```

Comment:
```suggestion
	public boolean playerDisablesFlight(Town town, Resident resident){
		if (!resident.hasTown()) {
			return Settings.flightDisableBy.equals("NEUTRAL");
		}

		Town outsiderTown = resident.getTownOrNull();

		if (Settings.flightDisableBy.equals("ALLY")){
			return CombatUtil.isAlly(town, outsiderTown);
		}

		if(Settings.flightDisableBy.equals("ENEMY")) {
			return CombatUtil.isEnemy(town, outsiderTown);
		}

		return false;
	}
```

I believe this will achieve what you want.